submission/proxy_test.go

```diff
@@ -121,3 +122,48 @@ func TestProxyRefreshRootsErr(t *testing.T) {
 		}
 	}
 }
+
+func TestCTPolicyTypeFromString(t *testing.T) {
+	testCases := []struct {
+		name      string
+		from      string
+		wantPlc   CTPolicyType
+		errRegexp *regexp.Regexp
+	}{
+		{
+			name:      "chrome",
+			from:      "chrome",
+			wantPlc:   ChromeCTPolicy,
+			errRegexp: nil,
+		},
+		{
+			name:      "apple",
+			from:      "apple",
+			wantPlc:   AppleCTPolicy,
+			errRegexp: nil,
+		},
+		{
+			name:      "noValue",
+			from:      "crome",
+			wantPlc:   ChromeCTPolicy,
+			errRegexp: regexp.MustCompile("CTPolicyTypeFromString does not support value"),
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			gotPlc, err := CTPolicyTypeFromString(tc.from)
+			if gotErr, wantErr := err != nil, tc.errRegexp != nil; gotErr != wantErr {
+				var unwantedErr string
+				if gotErr {
```

Comment:
nit: Remove `unwantedErr`, and just unconditionally add %v of `err` to the `t.Errorf` below?